### PR TITLE
[DOC] Pivot Table

### PR DESCRIPTION
**Pull request by ajdapretnar**
##### Issue

Pivot Table needs docs.
#3823

##### Description of changes
Add documentation for the widget.

##### Includes
- [ ] Code changes
- [ ] Tests
- [X] Documentation

### CODE REVIEW

File: doc/visual-programming/source/widgets/data/pivot.md [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+Reshape data table based on column values.
+
+**Inputs**
+
+- Data: input data set
+
+**Outputs**
+
+- Pivot Table: contingency matrix as set in the widget
```
Comment (janezd) [inline doc/visual-programming/source/widgets/data/pivot.md]: ... as showed? as seen? showed?

File: doc/visual-programming/source/widgets/data/pivot.md [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+- Filtered Data: subset selected from the plot
+- Grouped Data: data table grouped by row values
```
Comment (janezd) [inline doc/visual-programming/source/widgets/data/pivot.md]: Maybe `- Grouped Data: aggregates over groups defined by row values`?

File: doc/visual-programming/source/widgets/data/pivot.md [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+**Pivot Table** summarizes the data of a more extensive table into a table of statistics. The statistics can include sums, averages, counts, etc. The widget also allows selecting a subset from the plot and grouping by row values, which have to be a discrete variable. Data with only numeric variables cannot be displayed in the plot.
```
Comment (janezd) [inline doc/visual-programming/source/widgets/data/pivot.md]: `plot`? Probably `table`? (This appears twice.)

File: doc/visual-programming/source/widgets/data/pivot.md [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+![](images/Pivot-stamped.png)
+
+1. Discrete or numeric variable that will be used for row values. Numeric variables are considered as integers in this case. Variable values will appear as rows in the table.
```
Comment (janezd) [inline doc/visual-programming/source/widgets/data/pivot.md]: The last sentence is perhaps redundant. You can also remove `"in this case."` Also perhaps `that will be`.

File: doc/visual-programming/source/widgets/data/pivot.md [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+2. Discrete variable that will be used for column values. Variable values will appear as columns in the table.
+3. Values that will be used for aggregation. Aggregated values will appear as cells in the table.
```
Comment (janezd) [inline doc/visual-programming/source/widgets/data/pivot.md]: Consider removing `that will be`; also above.

File: doc/visual-programming/source/widgets/data/pivot.md [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+4. Aggregation methods:
+   - For any variable type:
+      - *Count*: number of instances that appear in the data
```
Comment (janezd) [inline doc/visual-programming/source/widgets/data/pivot.md]: `- *Count*: size of the group, that is, the number of instances with the given row and column value.`

I'm not totally sure this is better, though. :)

File: doc/visual-programming/source/widgets/data/pivot.md [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+      - *Count defined*: number of non-empty (not NaN) instances in the data.
```
Comment (janezd) [inline doc/visual-programming/source/widgets/data/pivot.md]: Huh, maybe "number of instances with this combination of the row and column value, for which the value that is used for aggregation is defined".

File: doc/visual-programming/source/widgets/data/pivot.md [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+   - For discrete variables:
+      - *Majority*: most frequent value of the subset.
+5. Tick the box on the left to automatically output any changes. Alternatively, press *Apply* .
+
+Discrete variables
+------------------
+
+![](images/Pivot-discrete.png)
+
+Example of a pivot table with only discrete variables selected. We are using *heart-disease* data set for this example. We are using the values of *diameter narrowing* as row values, namely 0 and 1. Our columns are values of *gender*, namely female and male. We are using *thal* as values in our cells.
```
Comment (janezd) [inline doc/visual-programming/source/widgets/data/pivot.md]: `We are using the values of *diameter narrowing* as row values` -> `Rows correspond to values of *diameter narrowing* variable`.

You can skip `namely`.

File: doc/visual-programming/source/widgets/data/pivot.md [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+We have selected *Count* and *Majority* as aggregation methods. In the pivot table, we can see the number of instances that do not have diameter narrowing and are female. There are 72 such patients. Concurrently, there are 92 male patients that don't have diameter narrowing. Thal values don't have any effect here, we are just counting occurrences in the data.
+
+The second row shows majority. This means most female patients that don't have diameter narrowing have normal thal results. Conversely, female patients that have diameter narrowing most often have reversable defect.
+
+Numeric variables
+-----------------
+
+![](images/Pivot-continuous.png)
+
+Example of a pivot table with numeric variables. We are using *heart-disease* data set for this example. We are using the values of *diameter narrowing* as row values, namely 0 and 1. Our columns are values of *gender*, namely female and male. We are using *rest SBP* as values in our cells.
```
Comment (janezd) [inline doc/visual-programming/source/widgets/data/pivot.md]: Same as above.

File: doc/visual-programming/source/widgets/data/pivot.md [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+
+We have selected *Count*, *Sum* and *Median* as aggregation methods. Under *Count*, we see there are 72 female patients that don't have diameter narrowing, same as before for discrete values. What is different are the sum and median aggregations. We see that the sum of resting systolic blood pressure for female patients that don't have diameter narrowing is 9269 and the median value is 130.
+
+Example
+-------
+
+We are using *Forest Fires* for this example. The data is loaded in the [Datasets](../data/datasets.md) widget and passed to **Pivot Table**. *Forest Fires* datasets reports forest fires by the month and day they happened. We can aggregate all occurrences of forest fires by selecting *Count* as aggregation method and using *month* as row and *day* as column values. Since we are using *Count*, it does not matter what our *Values* variable will be, so we will leave it as is.
```
Comment (janezd) [inline doc/visual-programming/source/widgets/data/pivot.md]: `it does not matter what our *Values* variable will be` -> maybe `*Values* is unimportant` (or something similar)

File: doc/visual-programming/source/widgets/data/pivot.md (hunk 1 above)
Comment (ajdapretnar) [reply doc/visual-programming/source/widgets/data/pivot.md]: shown :)

**Comment by ajdapretnar** [comment]
Comments addressed as well as possible.

**Pull request merged**